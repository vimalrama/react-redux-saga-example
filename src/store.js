import { createStore, applyMiddleware } from 'redux';
import { connectRouter, routerMiddleware } from 'connected-react-router';
import createHistory from 'history/createBrowserHistory';
import { composeWithDevTools } from 'redux-devtools-extension';
import createSagaMiddleware from 'redux-saga';

import rootReducer from './reducers';
import sagas from './sagas';

export const history = createHistory();

const sagaMiddleware = createSagaMiddleware();

const initialState = {};
const enhancers = [];
const middleware = [
  routerMiddleware(history),
  sagaMiddleware
];

// if (process.env.NODE_ENV === 'development') {
//   const devToolsExtension = window.__REDUX_DEVTOOLS_EXTENSION__;

//   if (typeof devToolsExtension === 'function') {
//     enhancers.push(devToolsExtension());
//   }
// }

const composedEnhancers = composeWithDevTools(
  applyMiddleware(...middleware),
  ...enhancers
);

const store = createStore(
  connectRouter(history)(rootReducer),
  initialState,
  composedEnhancers
);

sagaMiddleware.run(sagas);

export default store;
