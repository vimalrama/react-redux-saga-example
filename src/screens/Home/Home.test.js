import React from 'react';
import renderer from 'react-test-renderer';
import HomeComponent from './index';

import store from '../../store';

it('Testing Home Component', () => {
  const tree = renderer
    .create(<HomeComponent store={store} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
