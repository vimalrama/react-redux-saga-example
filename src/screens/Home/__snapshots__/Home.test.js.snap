// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Home Component 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "marginLeft": 40,
      "marginRight": 40,
    }
  }
>
  <button
    className="ui teal button"
    onClick={[Function]}
    role="button"
  >
    Load Users
  </button>
</div>
`;
